### PR TITLE
Add transparent (for now) babel step to JS app compilation

### PR DESCRIPTION
**Pull request by sndrs**
## What does this change?

- reopens #15130 
- adds babel 'transpilation' step to app compilation, but as yet no actual plugins
- webpack build uses this compiled code currently, but clearly we wouldn't do this once webpack is 100%.

## What is the value of this and can you measure success?

first step towards ES6 in frontend

## Does this affect other platforms - Amp, Apps, etc?

no

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
-        },
+        "production": {},
         "development": {
             "plugins": [
                 "transform-async-to-generator"
             ]
+        },
+        "admin": {
+            "presets": [
+                "latest"
```
Comment (gtrufitt) [inline .babelrc]: I know this is admin, but maybe it's worth using https://github.com/babel/babel-preset-env so we can specify browser versions (like we do with autoprefixer).

File: tools/__tasks__/compile/javascript/babel.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+const execa = require('execa');
+
+const {src, transpiled} = require('../../config').paths;
+
+module.exports = {
+    description: 'Transpile',
+    task: () => execa('babel', [`${src}/javascripts`, '-d', `${transpiled}/javascripts`, '--ignore', 'bower_components/,components/,vendor/'], {
```
Comment (gtrufitt) [inline tools/__tasks__/compile/javascript/babel.js]: Worth using `--out-dir` instead of `-d` to increase comprehension?

**Review by gtrufitt — APPROVED** [review]

**Review by gtrufitt — APPROVED** [review]
LGTM 👌

File: .babelrc [hunk 3]
```diff
@@ -1,14 +1,19 @@
...
         "development": {
             "plugins": [
                 "transform-async-to-generator"
             ]
+        },
+        "admin": {
+            "presets": [
+                ["env", {
+                    "targets": {
+                        "browsers": ["last 2 Chrome versions, last 1 Safari version, last 2 Firefox versions, last 2 Edge versions"]
```
Comment (SiAdcock) [inline .babelrc]: Does this go back far enough? Does this mean that once we start using ES6 <=IE11 will go boom?

**Comment by SiAdcock** [comment]
Why not use [`babel-loader`](https://github.com/babel/babel-loader) in the Webpack config, rather than loading from the `transpiled` directory? Arguably, this is what we'll be doing when we phase out `r.js`

File: .babelrc (hunk 3 above)
Comment (sndrs) [inline .babelrc]: only on admin

**Comment by sndrs** [comment]
> Why not use babel-loader in the Webpack config, rather than loading from the transpiled directory? Arguably, this is what we'll be doing when we phase out r.js

I didn't want to get in the way of any work you're doing there. i can update that if you like?

**Comment by SiAdcock** [comment]
No, you're right @sndrs, I can do that as a separate PR 😄

**Comment by sndrs** [comment]
got a better plan. closing for a second time

**Pull request closed without merging**